### PR TITLE
Extend test event processor with sequence numbers validation

### PR DESCRIPTION
**Pull request by cristipogacean**
add the logic for validation of incrementally reported sequence numbers in the test event telemetry processor

### CODE REVIEW

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs [hunk 1]
```diff
@@ -68,5 +68,20 @@ public class StopResult : IResult
         /// Indicates the number of duplicate messages detected by IncrementalIntValueChecker.
         /// </summary>
         public uint DuplicateValueCount { get; set; }
+
+        /// <summary>
+        /// Indicates the number of dropped sequence numbers detected by SequenceNumberChecker.
+        /// </summary>
+        public uint DroppedSequenceCount { get; set; }
```
Comment (karok2m) [inline tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs:75]: Can you please add those fields into `StopResult` model that we have in E2E project as well ?

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs [hunk 2]
```diff
@@ -40,6 +40,7 @@ public class TelemetryValidator : ITelemetryValidator
         private ValueChangeCounterPerNodeId _valueChangeCounterPerNodeId;
         private MissingValueChangesChecker _missingValueChangesChecker;
         private IncrementalIntValueChecker _incrementalIntValueChecker;
+        private SequenceNumberChecker _incrementalSequenceChecker;
```
Comment (dacolgit) [inline tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs:43]: just an observation that is not strictly related to this pr. 
In this file there is a mix of coding styles regarding curly brackets. Sometimes the bracket is on the new line:
`public async Task<StartResult> StartAsync(ValidatorConfiguration configuration)`
`{`
sometimes is on the same line:
`if (_cancellationTokenSource != null) {`

I would like to know what do you think. Should we have to standardize the style when we made some changes to a file?

**Review by dacolgit — DISMISSED** [review]

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs (hunk 1 above)
Comment (cristipogacean) [reply tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs:75]: done

Comment (karok2m) [reply tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs:75]: Here is the file path for that one: `e2e-tests\IIoTPlatform-E2E-Tests\TestEventProcessor\StopResult.cs`

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/Checkers/SequenceNumberChecker.cs [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                    _latestValue = curValue;
+                    return;
+                }
+
+                if (curValue == _latestValue + 1) {
+                    _latestValue = curValue;
+                    return;
+                }
+
+                if (curValue == _latestValue) {
```
Comment (karok2m) [inline tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/Checkers/SequenceNumberChecker.cs:52]: Both of previous if conditions assign `curValue` to `_latestValue`. Looks like they are going to cause duplicate detection here.

**Review by karok2m — APPROVED** [review]

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs (hunk 2 above)
Comment (cristipogacean) [reply tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/TelemetryValidator.cs:43]: I agree, we should be consistent with the coding guideline.

File: tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs (hunk 1 above)
Comment (cristipogacean) [reply tools/e2etesting/TestEventProcessor/TestEventProcessor.Businesslogic/StopResult.cs:75]: found it.

**Pull request merged**